app/views/layouts/_header.html.erb

```diff
@@ -17,19 +17,17 @@
       <li> <%= link_to "Cart", cart_path %> </li>
 
       <% if current_user %>
-        <% if current_admin? %>
+        <% if store_owner? %>
```

Comment:
So here we should make a decision about my last comment... Currently, store_owner? checks for current_user's presence (although I argued that it might be better on the user model and therefore not do that)... if we keep it the way it is, this 'if store_owner?' line doesn't need to nest inside the 'if current_user' if statement. If could just look like this:

show dashboard if store_owner?
show profile if current_user else
show <guest links> if !current_user

Then there's no nesting. Then again, if you make change I suggested - moving store_owner? to the user model - this nesting would make more sense.